evercrypt_backend/src/provider.rs

```diff
@@ -296,9 +296,56 @@ impl OpenMlsCrypto for EvercryptProvider {
         .map_err(|_| CryptoError::HpkeDecryptionError)
     }
 
+    fn hpke_setup_sender_and_export(
+        &self,
+        config: HpkeConfig,
+        pk_r: &[u8],
+        info: &[u8],
+        exporter_context: &[u8],
+        exporter_length: usize,
+    ) -> Result<(KemOutput, ExporterSecret), CryptoError> {
+        let hpke = Hpke::<HpkeEvercrypt>::new(
+            hpke::Mode::Base,
+            kem_mode(config.0),
+            kdf_mode(config.1),
+            aead_mode(config.2),
+        );
+        let (kem_output, context) = hpke
+            .setup_sender(&pk_r.into(), info, None, None, None)
+            .map_err(|_| CryptoError::SenderSetupError)?;
+        let exported_secret = context
+            .export(exporter_context, exporter_length)
+            .map_err(|_| CryptoError::ExporterError)?;
+        Ok((kem_output, exported_secret))
+    }
+
+    fn hpke_setup_receiver_and_export(
+        &self,
+        config: HpkeConfig,
+        enc: &[u8],
+        sk_r: &[u8],
+        info: &[u8],
+        exporter_context: &[u8],
+        exporter_length: usize,
+    ) -> Result<ExporterSecret, CryptoError> {
+        let hpke = Hpke::<HpkeEvercrypt>::new(
+            hpke::Mode::Base,
+            kem_mode(config.0),
+            kdf_mode(config.1),
+            aead_mode(config.2),
+        );
+        let context = hpke
+            .setup_receiver(enc, &sk_r.into(), info, None, None, None)
```

Comment:
Looking at the `&sk_r.into()` I feel like hpke-rs should have a byte slice API as well. 🤔

Reply:
It would be a bit more generic. But it wouldn't make that much of a difference, I guess?

Reply:
It avoids some allocations. But it's unrelated to this PR really 😄